tasks/libs/common/github_api.py

```diff
@@ -88,15 +89,24 @@ def make_request(self, path, headers=None, method="GET", data=None, json_output=
         headers["Authorization"] = f"token {self.api_token}"
         headers["Accept"] = "application/vnd.github.v3+json"
 
-        return self.request(
-            path=path,
-            headers=headers,
-            data=data,
-            json_input=False,
-            json_output=json_output,
-            stream_output=False,
-            method=method,
-        )
+        for retry_count in range(self.requests_500_retry_count):
+            try:
+                return self.request(
+                    path=path,
+                    headers=headers,
+                    data=data,
+                    json_input=False,
+                    json_output=json_output,
+                    stream_output=False,
+                    method=method,
+                )
+            except APIError as e:
+                if 500 <= e.status_code < 600:
+                    # We wait progressively more at each retry in case of overloaded servers
+                    time.sleep(self.requests_sleep_time + retry_count * self.requests_sleep_time)
+                else:
+                    raise e
+        raise Exit(message=f"Failed while making HTTP request: {method} {path}", code=1)
```

Comment:
Could the retry logic be put in the `request` method of the `RemoteAPI` class directly? That way you wouldn't have to copy-paste it on all classes that derive from `RemoteAPI`.